OneSignalSDK/unittest/src/test/java/com/onesignal/ShadowOSUtils.java

```diff
@@ -7,9 +7,15 @@
 @Implements(OSUtils.class)
 public class ShadowOSUtils {
 
+   public static boolean isRunningOnMainThread;
+
    public static String carrierName;
    public static int subscribableStatus;
 
+   public static boolean isRunningOnMainThread() {
```

Comment:
do we need this to be shadowed? shoudn't the test fail if it is running on the main thread? in that way we can fix it

Reply:
UnitTests always think we are on Main Thread and will fail due to the RuntimeException, so now we have a shadow so we can easily test when we want to assume Main Thread or not

Reply:
yeah, but this shadow is being used in many places, so we are hiding tests that we actually want to check if it is not running on the main thread, josh pointed this on the ANR fix https://github.com/OneSignal/OneSignal-Android-SDK/pull/1099#discussion_r460240990

so if there are tests that are failing due to this runtime exception then we should make those tests don't run on the main thread